src/components/Layout/MainHeader/MainHeader.tsx

```diff
@@ -1,30 +1,77 @@
+import { useState } from "react";
+
 import { NavLink } from "react-router-dom";
 import { navKeys } from "../../../routes/routes";
 
-import { StyledMainHeader, MainHeadingBrandName } from "./MainHeaderStyles";
-import { Button, Container } from "../../../common/styles/componentsStyles";
-
 import logo from "../../../assets/logo.png";
-import { Cart } from "../../../assets/icons";
+import {
+  CartIcon,
+  CloseIcon,
+  MobileMenuIcon,
+  OrderHistoryIcon,
+} from "../../../assets/icons";
+
+import {
+  StyledMainHeader,
+  MainHeadingBrandName,
+  CartButton,
+  ResponsiveButton,
+} from "./MainHeaderStyles";
+import { Container } from "../../../common/styles/componentsStyles";
+
+import useWindowDimensions from "../../../hooks/useWindowDimensions";
+import MobileMenu from "./MobileMenu";
 
 const MainHeader = () => {
-  return (
-    <StyledMainHeader>
-      <NavLink to="/">
-        <img className="logo" alt="" src={logo} />
-      </NavLink>
-      <MainHeadingBrandName>Candy&nbsp;Street</MainHeadingBrandName>
+  const { width } = useWindowDimensions();
+  const [menuVisible, setMenuVisible] = useState<boolean>(false);
+
+  const toggleMenuHandler = () => {
+    setMenuVisible((prevState: boolean) => !prevState);
+  };
+
+  let navButtons;
+
+  //desktop and tablet versions
+  if (width >= 540) {
+    navButtons = (
       <Container>
         <NavLink to={navKeys.history}>
-          <Button>Order History</Button>
+          <ResponsiveButton>
+            {width > 920 ? <span>OrderHistory</span> : <OrderHistoryIcon />}
+          </ResponsiveButton>
         </NavLink>
         <NavLink to={navKeys.cart}>
-          <Button>
-            <Cart />
-            Cart
-          </Button>
+          <CartButton>
+            <CartIcon />
+            {width > 1200 ? <span>Cart</span> : null}
+          </CartButton>
         </NavLink>
       </Container>
+    );
+  }
+
+  //mobile version
+  if (width < 540) {
+    navButtons = (
+      <Container>
+        <ResponsiveButton onClick={toggleMenuHandler}>
+          {menuVisible ? <CloseIcon /> : <MobileMenuIcon />}
+        </ResponsiveButton>
+      </Container>
+    );
+  }
+
+  return (
+    <StyledMainHeader>
+      <NavLink to="/">
+        <img className="logo" alt="" src={logo} />
+      </NavLink>
+      <MainHeadingBrandName>Candy&nbsp;Street</MainHeadingBrandName>
+      {navButtons}
+      {width < 540 && menuVisible && (
```

Comment:
MAGIC NUMBER. it's better to use a variable with a specific name, or a constant preferably consistent that's coming from your theme for example